### DISCUSSION TITLE
GeographicLib

### DISCUSSION THREAD

**Discussion by jonsberndt** [JSBSim-Team/jsbsim#524] [General]
I just noticed the inclusion of GeographicLib - in particular for calculating waypoint characteristics. Very nice. Does the fact that it uses a different license cause any issues?

**Comment by seanmcleod** [comment]
So JSBSim is using the GNU Lesser General Public License (GNU LGPL) and GeographicLib is using the more permissive MIT/X11 license.

From the GNU licenses FAQ - https://www.gnu.org/licenses/gpl-faq.en.html

> The X11 license is compatible with the GPL, so you can add a module to the GPL-covered program and put it under the X11 license. But if you were to incorporate them both in a larger program, that whole would include the GPL-covered part, so it would have to be licensed as a whole under the GNU GPL.

So I don't see any specific issue.

**Comment by jonsberndt** [comment]
Seems to be an error in the call to GetDistanceTo (from FGWaypoint). I don't have time to debug this, but I'm stepping through the code with:

Target lat: -1.982100
Target lon: 0.700800

and calling GetDistanceTo from within FGWayPoint and get a floating point exception:

cout << "Target lat: " << target_latitude_rad << endl;
cout << "Target lon: " << target_longitude_rad << endl;
    double wp_distance = source.GetDistanceTo(target_longitude_rad,
                                              target_latitude_rad);

Maybe I can look at this later, but it would be better if someone familiar with the GeographicLib could take a look. Are negative numbers supported? Are there any limitations to the code? Is the great circle distance returned?

**Comment by seanmcleod** [comment]
Take a look at this issue - https://github.com/JSBSim-Team/jsbsim/issues/475

What compiler are you using? 

You mention the target's coordinates, what were are the source's coordinates? Are source and target extremely close?

**Reply by jonsberndt** [reply]
The source coords are just the current location. I guess since this is a first-frame problem I should print out the coords and verify that they are good. That could well be the issue: uninitialized coordinates. Here's my XML:
```xml
		<waypoint_distance name="system/distance-to-dugway" unit="FT">
			  <target_latitude unit="RAD">  location/site_long_rad </target_latitude>
			  <target_longitude unit="RAD"> location/site_lat_rad </target_longitude>
			  <source_latitude unit="RAD">  position/lat-geod-rad </source_latitude>
			  <source_longitude unit="RAD"> position/long-gc-rad </source_longitude>
		</waypoint_distance>	
```

**Reply by jonsberndt** [reply]
Well, the current location is fine.

**Reply by jonsberndt** [reply]
Further info: compilation is being done under Cygwin 64 on a Windows 10 machine.

**Reply by jonsberndt** [reply]
Ugh. I'm an idiot.
```xml
	<waypoint_distance name="system/distance-to-dugway" unit="FT">
		  <target_**latitude** unit="RAD">  location/site_**long**_rad </target_latitude>
		  <target_**longitude** unit="RAD"> location/site_**lat**_rad </target_longitude>
		  <source_latitude unit="RAD">  position/lat-geod-rad </source_latitude>
		  <source_longitude unit="RAD"> position/long-gc-rad </source_longitude>
	</waypoint_distance>	
```
Maybe there should be some kind of a bounds check in the code for idiots? LoL.

**Reply by bcoconni** [reply]
> Maybe there should be some kind of a bounds check in the code for idiots? LoL.

Well, JSBSim should never crash even when supplied with silly data. Could you please fill a bug report ?
